### DISCUSSION TITLE
Are address(0) checks worth it for msg.sender and similar?

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/evm-monorepo#38] [General]
Are `address(0)` checks worth it for `msg.sender` and similar?

1. There's little to no change that senders are going to be `address(0)`
2. Assumptions can still be broken by easy cases such as `address(1)`

These are still good for recipients (for e.g. nullable fallbacks in clients).

**Comment by PaulRBerg** [comment]
I have been pondering this ever since we discussed the idea last week .. and I'm still of the opinion that the checks are worth it, at least for the `sender`.

My rationale is based on nothing else but the precautionary principle. The zeroness of the `sender` struct member is used to check whether a stream exists or not. But allowing streams to be created from the zero address would break that assumption (so only *some* assumptions are broken by using `address(1)`).

Obviously, the risk of someone breaking into `address(0)` is infinitesimally low. That's not my worry.

We intend to deploy Sablier to all sorts of EVM-compatible chains. Not all of those chains have the same security guarantees as Ethereum L1. If we ever end up deploying Sablier to private chains (I don't see why not), then the network administrator could, in principle, make calls from the zero address. And that would have all sorts of consequences for the Sablier V2 protocol, because the protocol assumes that when the sender is zero, the stream does not exist.

**Comment by PaulRBerg** [comment]
Based on the rationale above and @razgraf's soft confirmation by not following up, I am locking this discussion.

**Reply by razgraf** [reply]
It was an open question, I don't think a follow-up was required from me.

From a [quick search on OZ](https://forum.openzeppelin.com/t/removing-address-0x0-checks-from-openzeppelin-contracts/2222), checks for `address(0)` are simply part of existing ERCs, some of which use `0x0` as common error-code type of responses. If we're using ERC721 it might be better to have these checks in place for certain cases.

**Comment by PaulRBerg** [comment]
While reviewing Flow, I noticed that you guys (@andreivladbrg, @smol-ninja) added a check to prevent zero addresses for the sender.

That check isn't found in Lockup, and I can't remember why. Based on my reading of this discussion, it seems that I have concluded that a check is worth it.

Perhaps we removed in order to optimize for gas efficiency and lower contract size?

**Edit**: as per this PR https://github.com/sablier-labs/v2-core/pull/196, part of the reason was indeed gas efficiency, but also redundancy - the zero address check used to be a filter for null streams.
